taichi/program/program.h

```diff
@@ -124,6 +125,7 @@ class TI_DLL_EXPORT Program {
   ~Program();
 
   CompileConfig &this_thread_config() {
+    std::lock_guard<std::mutex> _(config_mut);
```

Comment:
This might be a potentially perf hit. Is it possible to separate read and write so that you can only guard the writes?

Reply:
No, we can't. It still fails from time to time if we only guard the writes. However we can use a reader writer lock to solve this.